### ISSUE TITLE
False-positive result on change parameter type from `array` to `iterable`

### ISSUE THREAD

**Issue by vjik** [Roave/BackwardCompatibilityCheck#317]
Example:

```php
public static function map(array $array, $from, $to, $group = null): array
```

changed to:

```php
public static function map(iterable $array, $from, $to, $group = null): array
```

result will be:

```
[BC] CHANGED: The parameter $array of Yiisoft\Arrays\ArrayHelper::map() changed from array to iterable
```

Real example: https://github.com/yiisoft/arrays/pull/97

**Comment by Majkl578** [comment]
Not a false positive, changing `array` to `iterable` in non-final class/method will break all consumers overriding that method - parameters are contravariant and as such child types can't be more specific: https://3v4l.org/BsFuX

**Comment by vjik** [comment]
Thank you for the explanation!
I lost sight of the fact that class not final.

**Comment by Ocramius** [comment]
Can we verify if this affects also `final` classes? Agree on what @Majkl578 said here: on `non-final` classes, changing parameter types is generally not possible.

 * removing parameter types breaks child classes
 * widening parameter types breaks child classes
 * restricting parameter types breaks consumers
